go/api/cluster_client.go

```diff
@@ -0,0 +1,18 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// RedisClusterClient is a client used for connection to cluster Redis servers.
+type RedisClusterClient struct {
+	*baseClient
+}
+
+// NewRedisClusterClient creates a [RedisClusterClient] in cluster mode using the given [RedisClusterClientConfiguration].
+func NewRedisClusterClient(config *RedisClusterClientConfiguration) (*RedisClusterClient, error) {
```

Comment:
Please note that we changed the terminology in our code base. `RedisClusterClient` is now `GlideClusterClient`
`RedisClient` -> `GlideClient` 

Please align the code with the rest of the bindings (have a look at the Java binding for reference)

Reply:
Done